### ISSUE TITLE
[qt5-base:x64-windows, qt5-base:x86-windows] build failure

### ISSUE THREAD

**Issue by Natr1x** [microsoft/vcpkg#16374]
**Host Environment**
- OS: Windows 10 Pro, Version 20H2 (OS build: 19042.804 Experience: Windows Feature Experience Pack 120.2212.551.0)
- Compiler: revision

**To Reproduce**
Run the below command line from a fresh vcpgk installation root (although the below output is with all dependencies installed):
`PS C:\src\vcpkg> .\vcpkg.exe install --triplet=x64-windows qt5-base`

**Failure logs** 
- Command line output
  `PS C:\src\vcpkg> .\vcpkg.exe install --triplet=x64-windows qt5-base`
```
Computing installation plan...
warning: vcpkg appears to be in a Visual Studio prompt targeting x86 but is installing packages for x64-windows. Consider using --triplet x86-windows or --triplet x86-uwp.
The following packages will be built and installed:
    qt5-base[core]:x64-windows -> 5.15.2#1
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\osksod\AppData\Local\vcpkg\archives\12\1248760b6d03d65046b2c865036d6a2eb4f08d87.zip
Starting package 1/1: qt5-base:x64-windows
Building package qt5-base[core]:x64-windows...
-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: win32-msvc
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached C:/src/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Cleaning sources at C:/src/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/src/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/icu.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/zstdd.patch
-- Applying patch patches/mysql_plugin_include.patch
-- Applying patch patches/mysql-configure.patch
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Using source at C:/src/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x64-windows-dbg
-- Configuring x64-windows-dbg done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:144 (message):
    Command failed: C:/src/vcpkg/downloads/tools/jom/jom-1.1.3/jom.exe /J 5
    Working Directory: C:/src/vcpkg/buildtrees/qt5-base/x64-windows-dbg
    See logs for more information:
      C:\src\vcpkg\buildtrees\qt5-base\build-x64-windows-dbg-out.log
      C:\src\vcpkg\buildtrees\qt5-base\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/install_qt.cmake:92 (vcpkg_execute_build_process)
  ports/qt5-base/portfile.cmake:348 (install_qt)
  scripts/ports.cmake:133 (include)


Error: Building package qt5-base:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-base:x64-windows
  Vcpkg version: 2021-01-13-768d8f95c9e752603d2c5901c7a7c7fbdb08af35
```  
  `PS C:\src\vcpkg> .\vcpkg.exe install --triplet=x86-windows qt5-base`
  ```
PS C:\src\vcpkg> .\vcpkg.exe install qt5-base
Computing installation plan...
The following packages will be built and installed:
    qt5-base[core]:x86-windows -> 5.15.2#1
Detecting compiler hash for triplet x86-windows...
Could not locate cached archive: C:\Users\osksod\AppData\Local\vcpkg\archives\b6\b666a8f8b2245397d2e0b897585137f42ea5f1a0.zip
Starting package 1/1: qt5-base:x86-windows
Building package qt5-base[core]:x86-windows...
-- Figuring out qt target mkspec. Target arch x86
-- Target mkspec set to: win32-msvc
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached C:/src/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Cleaning sources at C:/src/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/src/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/icu.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/zstdd.patch
-- Applying patch patches/mysql_plugin_include.patch
-- Applying patch patches/mysql-configure.patch
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Using source at C:/src/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:144 (message):
    Command failed: C:/src/vcpkg/downloads/tools/jom/jom-1.1.3/jom.exe /J 5
    Working Directory: C:/src/vcpkg/buildtrees/qt5-base/x86-windows-dbg
    See logs for more information:
      C:\src\vcpkg\buildtrees\qt5-base\build-x86-windows-dbg-out.log
      C:\src\vcpkg\buildtrees\qt5-base\build-x86-windows-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/install_qt.cmake:92 (vcpkg_execute_build_process)
  ports/qt5-base/portfile.cmake:348 (install_qt)
  scripts/ports.cmake:133 (include)


Error: Building package qt5-base:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-base:x86-windows
  Vcpkg version: 2021-01-13-768d8f95c9e752603d2c5901c7a7c7fbdb08af35

Additionally, attach any relevant sections from the log files above.
```

- Log files:
[build-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/6029496/build-x64-windows-dbg-err.log)
[build-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6029497/build-x64-windows-dbg-out.log)
[build-x86-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/6029600/build-x86-windows-dbg-err.log)
[build-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6029601/build-x86-windows-dbg-out.log)

**Comment by Neumann-A** [comment]
Error 2 indicates a file not found error. The logs don't tell anything else so I would assume either you are running low on HDD space or your Virus protection removed `moc.exe`

**Comment by Natr1x** [comment]
> 
> 
> Error 2 indicates a file not found error. The logs don't tell anything else so I would assume either you are running low on HDD space or your Virus protection removed `moc.exe`

So moc.exe was indeed missing from `C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\bin` however it is still present in `C:\src\vcpkg\buildtrees\qt5-base\x86-windows-dbg\bin` but if the virus protection caused this then it might have messed up other things. Here is the output from `ls -Force -Recurse -Name .\qt5-base\` 
[qt5-base-files.txt](https://github.com/microsoft/vcpkg/files/6035535/qt5-base-files.txt)  

Though I guess it also shows that the issue with build qt5-base:x86-windows is different from when trying to build qt5-base:x64-windows. Please tell me if I should open a separate issue edit this one in any way.

There should not be any issue with space unless qt5-base requires more than 400gb.

**Comment by Natr1x** [comment]
OK, so new day and a new attempt to install in case there is a difference and lo and behold, *qt5-base:x86-windows* installs without trouble today. I still get the same errors when trying to install *qt5-base:x64-windows* though *moc.exe* is no longer missing in `C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\bin`

**Comment by JackBoosY** [comment]
@Natr1x Please note that any generated unsigned binary may be blocked by anti-virus.
